tests/e2e-inst-signatures/scripts/proctree_data_source.sh

```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/bash
```

Comment:
We should use something that can test childs and threads in a proper way. Something like:

```C
include <stdio.h>
#include <stdlib.h>
#include <pthread.h>
#include <unistd.h>
#include <sys/syscall.h>

#define SECS 360000

void* worker_function(void* arg) {

    pid_t pid = getpid();
    pid_t tid = syscall(SYS_gettid);
    pid_t ppid = getppid();

    printf("Thread (ppid=%d, pid=%d, tid=%d) started. Sleeping for %d seconds...\n", ppid, pid, tid, SECS);
    sleep(SECS);
    printf("Thread finished sleeping.\n");

    return NULL;
}

void* special_worker_function(void* arg) {

    pid_t pid = getpid();
    pid_t tid = syscall(SYS_gettid);
    pid_t ppid = getppid();

    printf("Thread (ppid=%d, pid=%d, tid=%d) started. Creating another thread...\n", ppid, pid, tid);

    pthread_t thread;
    int status;

    status = pthread_create(&thread, NULL, worker_function, NULL);
    if (status) {
        fprintf(stderr, "Error creating thread: %d\n", status);
    }

    status = pthread_join(thread, NULL);
    if (status) {
        fprintf(stderr, "Error joining thread: %d\n", status);
    }

    return NULL;
}

int main() {
    pthread_t thread1, thread2;
    int status;

    pid_t pid = getpid();
    pid_t tid = syscall(SYS_gettid);
    pid_t ppid = getppid();

    printf("Process (ppid=%d, pid=%d, tid=%d) started.\n", ppid, pid, tid);

    // Create first thread
    status = pthread_create(&thread1, NULL, worker_function, NULL);
    if (status) {
        fprintf(stderr, "Error creating thread 1: %d\n", status);
        exit(EXIT_FAILURE);
    }

    // Create second thread
    status = pthread_create(&thread2, NULL, special_worker_function, NULL);
    if (status) {
        fprintf(stderr, "Error creating thread 2: %d\n", status);
        exit(EXIT_FAILURE);
    }

    // Wait for first thread to finish
    status = pthread_join(thread1, NULL);
    if (status) {
        fprintf(stderr, "Error joining thread 1: %d\n", status);
        exit(EXIT_FAILURE);
    }

    // Wait for second thread to finish
    status = pthread_join(thread2, NULL);
    if (status) {
        fprintf(stderr, "Error joining thread 2: %d\n", status);
        exit(EXIT_FAILURE);
    }

    printf("All threads have finished.\n");
    return 0;
}
```

It tests:
- main process
- thread group leader
- a regular thread
- a thread that spawns another thread

We can later add a thread that calls exec.